### DISCUSSION TITLE
Tooling for creating decks such as this one

### DISCUSSION THREAD

**Discussion by jasonwilliams** [axelboc/anki-ultimate-geography#592] [General]
Hey @axelboc et al

I maintain https://github.com/jasonwilliams/anki and was curious to know if tooling such as this would be useful for helping to create larger decks. What is your current workflow? And is there anything that extension could do to improve it?

**Comment by axelboc** [comment]
Hey @jasonwilliams, looks like a very cool deck building tool!

Since our users are not all developers, we couldn't rely on a VS Code extension for installing/upgrading the deck (we use an Anki extension called [CrowdAnki](https://ankiweb.net/shared/info/1788670778) for that). Also, I don't think Markdown would be a god fit for us for writing cards, since we need to store structured data (including for translation purposes). Our current deck manager, [Brain Brew](https://github.com/ohare93/brain-brew) has a lot of powerful features -- amongst other things, it ensures that each note has a unique, stable ID, which allows upgrading the deck without losing progress.

Your extension seems really well suited for helping developers maintain their own decks. It provides a much faster workflow than our current set-up (build with BrainBrew, import with CrowdAnki). I might use it myself, actually! Creating decks directly in Anki is not fun, let's me honest... 😬

**Comment by aplaice** [comment]
I concur with @axelboc that the extension is really neat, though also agree with him regarding the drawbacks for use here, in particular!

In terms of general feedback/advice, the only thing that really comes to mind that in (in our case) about 90% of the time is probably spent on editing existing notes, so the future editing functionality would probably be valuable, for shared decks\*.  (People like to nit-pick. :))

\* for personal (unshared) decks it's probably not as crucial

**Comment by jasonwilliams** [comment]
This was very useful feedback, thanks!
I agree, for very structured data Markdown doesn’t really cut it, I suppose it’s more suited for taking quick notes and converting them to cards. I haven’t heard about Brain Brew but it’s certainly suited for a project as big as this one.

I think we may look into supporting more structured data in future to accommodate more medium-large size decks, but it’s great to hear you both would find it useful in its current state.

We are working on editing existing notes, this has been a difficult problem as the conversion from markdown to Anki/HTML is currently lossy, so we need to do it in such a way you can go back and forth.
